### DISCUSSION TITLE
Set the log level of `run` command

### DISCUSSION THREAD

**Discussion by eratio08** [quickwit-oss/quickwit#4377] [Q&A] [answered]
Hello, I'm new to using quickwit. I was wondering how I can set the log level of the quickwit service running in the Docker container? I could not find any information in the docs and searching the code I only found https://github.com/quickwit-oss/quickwit/blob/7458bbf31c38c116f8ed8892a8c6fed3949e637c/quickwit/quickwit-cli/src/main.rs#L66 which seems to always pick the default log level of info.

Is there any way to set the log level of the `run` command?

**Comment by trinity-1686a** [comment, marked as answer]
you can set the environment variable `RUST_LOG`, for instance in a shell you can run `RUST_LOG=debug quickwit run` (there are multiple ways to set environment variable depending on how you start your containers).
You can read more on what value you can put to get more precise control [on this page](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html#example-syntax).
I'm surprised we don't mention RUST_LOG anywhere in the documentation. We definitely should. I'll open a ticket for that (edit: https://github.com/quickwit-oss/quickwit/issues/4378)

**Reply by eratio08** [reply]
Is there a way to silence the planner output? It does not seem to be affected by the log level.

**Reply by trinity-1686a** [reply]
I'm not sure what you're calling "the planner". You can control verbosity per module, for instance if you don't want logs like
```
2024-01-16T11:04:07.474Z  INFO quickwit_janitor::actors::delete_task_pipeline: Spawning delete tasks pipeline. index_id=otel-logs-v0_6 root_dir=/home/trinity/dev/datasets/qwdata/delete_task_service
```
to show up, but still get info for everything else. You can set `RUST_LOG=info,quickwit_janitor::actors::delete_task_pipeline=warn` (or something higher than warn, such as error or off)

**Reply by eratio08** [reply]
So if I run the docker container I get a frequent `CURRENT PLAN`(and some more lines, mostly empty) output which is not affected by the `RUST_LOG`, which otherwise work perfect for all other output of the container. As its reporting about a plan I just called it planner output.

**Reply by trinity-1686a** [reply]
are you sure these are quickwit logs, and not logs inserted for instance by your container orchestration tool? I can't find any traces of messages looking like that in quickwit source code.

**Reply by eratio08** [reply]
Will try to get a sample of these logs. So it's definitely part of the container as it's local to the container logs of quickwit.

The orchestration I'm using is docker compose so there is no special logging injected by the compose orchestration.

**Reply by eratio08** [reply]
<img width="307" alt="image" src="https://github.com/quickwit-oss/quickwit/assets/5063643/4ee56b4a-eb1a-44e2-a4c7-74d5a2c2e889">
This is the regular output of the container. It happens very frequently.

**Reply by fulmicoton** [reply]
@eratio08 Which tag are you using? This is supposed to happen only on `qw-fulmicoton`.

**Reply by eratio08** [reply]
Actually I was not using any tag so pretty random I guess, I thing it defaults to latest, which will be random if it's not deliberately set. Which tag is recommended here? https://quickwit.io/docs/get-started/quickstart#use-quickwits-docker-image does not seem to name any.

Let me check which I've pulled 🔍

**Reply by eratio08** [reply]
I've pulled
```
quickwit/quickwit latest sha256:3e7d718c1104d058e29dc0f6e5c84facc3d993523deb9a7058be3b25392ef18f   aac8163a6ec0   7 days ago      292MB
```

I'm assuming using [0.7.0](https://hub.docker.com/layers/quickwit/quickwit/0.7.0/images/sha256-21c033ce27823fd9b3abec6e482cafaf5ed2039f2240822254ab8173c9e151d5?context=explore) would be proper.

**Comment by anila-intelops** [comment]
Hi all,
i am trying to instrument my application with open telemetry, i am trying to send my traces to quick wit. i have installed quick wit in cluster
```
func InitTracer() (*sdktrace.TracerProvider, error) {
    ctx := context.Background()

    headers := map[string]string{
        "service-name": "myapp",
    }

    client := otlptracegrpc.NewClient(
        otlptracegrpc.WithEndpoint("quickwit cluster installed url:port"),
        otlptracegrpc.WithHeaders(headers),
        otlptracegrpc.WithInsecure(),
    )

    exporter, err := otlptrace.New(ctx, client)
    if err != nil {
        log.Fatalf("failed to initialize exporter: %e", err)
    }

    res, err := resource.New(
        ctx,
        resource.WithAttributes(
            attribute.String("service.name", "myapp"),
            attribute.String("library.language", "go"),
        ),
    )
    if err != nil {
        log.Fatalf("failed to initialize resource: %e", err)
    }

    // Create the trace provider
    tp := sdktrace.NewTracerProvider(
        trace.WithSampler(trace.AlwaysSample()),
        sdktrace.WithBatcher(exporter),
        sdktrace.WithResource(res),
    )

    // Set the global trace provider
    otel.SetTracerProvider(tp)

    // Set the propagator
    propagator := propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{})
    otel.SetTextMapPropagator(propagator)

    return tp, nil
}
```
but i am getting below error
[GIN] 2024/01/17 - 11:07:53 | 200 |      487.21µs |             ::1 | GET      "/books"
2024/01/17 11:07:58 traces export: rpc error: code = Unimplemented desc = unexpected HTTP status code received from server: 404 (Not Found); transport: received unexpected content-type "text/plain; charset=utf-8"

**Reply by trinity-1686a** [reply]
can you open a separate discussion? This doesn't seem related to setting Quickwit log level

**Comment by fmassot** [comment]
@eratio08 which docker image are you using? having those logs is not normal

**Reply by fulmicoton** [reply]
This is from my tag qw-fulmicoton. I assume @eratio08 got it by pulling the :latest tag

**Reply by eratio08** [reply]
Yep was the latest tag which might have randomly selected that image if it was published as the most recent during that time.
